### PR TITLE
[WIP] Akka typed examples

### PR DESCRIPTION
**Pull request by jlprat**
## Purpose

This PR adds examples on how Akka HTTP and Akka Typed interact with each other.
## References

References #2584

## Changes

Only the Scala example is there at the moment. Java one is missing.
There will be some build problems I suppose

### CODE REVIEW

File: docs/src/test/scala/docs/http/scaladsl/HttpServerWithTypedSpec.scala [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+
+    import akka.http.scaladsl.Http
+
+    import scala.concurrent.ExecutionContextExecutor
+    import scala.util.{ Success, Failure }
+
+    val system = ActorSystem[Done](Behaviors.setup[Done] { ctx =>
+      // http doesn't know about akka typed so create untyped system/materializer
+      implicit val untypedSystem: actor.ActorSystem = ctx.system.toUntyped
+      implicit val materializer: ActorMaterializer = ActorMaterializer()(ctx.system.toUntyped)
```
Comment (johanandren) [inline docs/src/test/scala/docs/http/scaladsl/HttpServerWithTypedSpec.scala]: There actually is an `akka-stream-typed` module with a `akka.stream.typed.scaladsl.ActorMaterializer` for this. Note entirely sure if that should go in the sample or not though.

**Comment by johanandren** [comment]
Should we merge this and follow up with a Java counterpart later?

**Comment by jrudolph** [comment]
> Should we merge this and follow up with a Java counterpart later?

Added #2664 to track this. Let us know, @jlprat, if you'd like to work on this as well.

**Comment by jlprat** [comment]
Hey, yes I'm up for it, but timewise I'm not sure when I will be able to take it.
So if someone else wants to take it first, feel free!

**Comment by Reeebuuk** [comment]
One question around this one. 
How do we test the routes via akka-http-testkit if routes are inside of the behavior?  I'm new to typed Akka so maybe this is a silly question.

**Pull request merged**